tests/test_other.py

```diff
@@ -5900,12 +5901,24 @@ def test_realpath(self):
 
   printf("Resolved: %s\n", t_realpath_buf);
   free(t_realpath_buf);
+
+  std::error_code ec;
+  std::filesystem::path filename = "/boot/README.txt";
+  using std::filesystem::perms;
+  std::filesystem::permissions(filename, perms::group_write, ec);
+  auto s = std::filesystem::status(filename, ec);
+  bool readonly = perms::none == (s.permissions() & (perms::owner_write | perms::group_write | perms::others_write));
+  if(readonly) {
+    perror("not writeable");
+    return 1;
+  }
```

Comment:
Can you move this to its own test?   It doesn't seem like its related to `realpath`.

Also, I wonder if you could write the test interm of the lower level C API (stat/fstat/fstatat) instead to avoid testing more than we need to?

I see some existing stat tests that could perhaps be modified:

```
$ git grep "def test_*stat" tests/*.py
tests/test_core.py:  def test_statics(self):
tests/test_core.py:  def test_stat(self):
tests/test_core.py:  def test_stat_chmod(self):
tests/test_core.py:  def test_stat_mknod(self):
tests/test_core.py:  def test_statvfs(self):
tests/test_core.py:  def test_static_variable(self):
tests/test_other.py:  def test_stat_silly(self):
tests/test_other.py:  def test_stat_fail_alongtheway(self):
```

I looks like maybe `tests/stat/test_chmod.c` would be good home this new test?  (Although I'm not sure why that test doesn't already fail if this fix is needed).

Reply:
I guess were not testing `fchmodat` as part of `tests/stat/test_chmod.c`?

Reply:
> Can you move this to its own test? It doesn't seem like its related to realpath.

sure i will. and let me check test_chmod to see whats the difference.

Reply:
true enough fchmodat wasnt tested and it failed the new C api test i created without the fix here